http/vibe/http/common.d

```diff
@@ -336,13 +337,222 @@ class HTTPStatusException : Exception {
 	string debugMessage;
 }
 
+final class MultiPartBodyPart
```

Comment:
I'd keep this named as `MultiPart` and rename the new `MultiPart` class below to `MultiPartBody`.

The idea for `HTTPClientRequest.writePart` was that it would indeed just write a single part, so that not all parts would have to be constructed up front. For this reason I'd also not expose the new `MultiPart` class below as a public API, but just use the code within to implement the `writePart` method with proper lazy writing of the preamble etc.